lib/streamlit/elements/widgets/button_group.py

```diff
@@ -168,6 +171,28 @@ def _build_proto(
 
 
 class ButtonGroupMixin:
+    @overload  # These overloads are not documented in the docstring, at least not at this time, on the theory that most people won't know what it means. And the Literals here are a subclass of int anyway.
+    def feedback(
+        self,
+        options: _FeedbackOptions2 = ...,
```

Comment:
When we use it like this, the autocompletion does not look too nice, because it is not obvious what `_FeedbackOptions2` is referring to. I have just realized that this is already an issue with the existing API where we use `_FeedbackOptions`. Perhaps we should use the `Literal` types directly for typing the arguments. This means a little bit more duplication, but at least the user-facing API is nicer

Reply:
Oh, interesting! Yeah, the only reason I kept any concept of a _FeedbackOptions around was because there are two other parts of the code that use it, so not having it be a variable is a DRY violation. But hopefully the typechecker will keep those two other instances from getting out of sync, so maybe it's not so bad. I'm happy to make the change if it improves autocompletion.

This, incidentally, calls into question line [290](https://github.com/streamlit/streamlit/blob/a7720e45a843caf91937fb68004d177c92b41601/lib/streamlit/elements/widgets/button_group.py#L290C8-L290C88): ` if not isinstance(options, list) and options not in get_args(_FeedbackOptions):`. I think the `not isinstance(options, list)` clause is completely unnecessary. If you pass any list in to this function it just, uh... doesn't generate a widget. So I think we should just remove this clause.

Reply:
yeah great catch, thanks! It looks like some left-over from an older state 😞 
With regards to the auto-completion question I'd like to hear the opinion from @snehankekre and @jrieke